pkg/service/connect_service.go

```diff
@@ -96,14 +98,44 @@ func (s *ConnectService) setupServer() (net.Listener, error) {
 		)
 	}
 	s.server = http.Server{
-		ReadTimeout:       2 * time.Second,
-		WriteTimeout:      4 * time.Second,
 		ReadHeaderTimeout: time.Second,
 		Handler:           handler,
 	}
 	return lis, nil
 }
 
+func (s *ConnectService) Notify(n NotificationType) {
+	for _, send := range s.subs {
+		send <- n
+	}
+}
+
+func (s *ConnectService) EventStream(
+	ctx context.Context,
+	req *connect.Request[emptypb.Empty],
+	stream *connect.ServerStream[schemaV1.EventStreamResponse],
+) error {
+	s.subs[req] = make(chan NotificationType, 1)
```

Comment:
do we need a mutex on this?

Reply:
added lock to surround the writing to and deleting of the channels

Reply:
the creation of the channel does not need a mutex as the pointer value will not be reused until after the defer function runs, so will be threadsafe